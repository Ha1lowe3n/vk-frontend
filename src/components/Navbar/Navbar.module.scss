.navbar {
    width: 164px;
    margin-right: 15px;
}
