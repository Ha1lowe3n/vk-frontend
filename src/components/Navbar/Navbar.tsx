import React from "react";

import styles from "./Navbar.module.scss";

import {
    NavbarCallIcon,
    NavbarFriendsIcon,
    NavbarMessageIcon,
    NavbarNewsIcon,
    NavbarProfileIcon,
} from "../Icons";
import { NavbarItem } from "./NavbarItem/NavbarItem";

export const Navbar: React.FC = () => {
    return (
        <nav className={styles.navbar}>
            <ul>
                <NavbarItem title={"My profile"}>
                    <NavbarProfileIcon />
                </NavbarItem>
                <NavbarItem title={"News"}>
                    <NavbarNewsIcon />
                </NavbarItem>
                <NavbarItem title={"Messages"}>
                    <NavbarMessageIcon />
                </NavbarItem>
                <NavbarItem title={"Calls"}>
                    <NavbarCallIcon />
                </NavbarItem>
                <NavbarItem title={"Friends"}>
                    <NavbarFriendsIcon />
                </NavbarItem>
            </ul>
        </nav>
    );
};
