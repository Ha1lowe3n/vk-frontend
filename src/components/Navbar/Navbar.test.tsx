import { render, screen } from "@testing-library/react";
import { Navbar } from "./Navbar";

it("render Navbar component", () => {
    render(<Navbar />);
    expect(screen.getByText("My profile")).toBeInTheDocument();
    expect(screen.getByText("News")).toBeInTheDocument();
    expect(screen.getByText("Messages")).toBeInTheDocument();
    expect(screen.getByText("Calls")).toBeInTheDocument();
    expect(screen.getByText("Friends")).toBeInTheDocument();

    expect(screen.getByRole("list")).toBeInTheDocument();
});

it("Navbar snapshot", () => {
    const view = render(<Navbar />);
    expect(view).toMatchSnapshot();
});
