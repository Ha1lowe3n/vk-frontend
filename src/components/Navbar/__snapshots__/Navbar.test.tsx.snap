// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar"
      >
        <ul>
          <li
            class="item"
          >
            <svg
              fill="none"
              height="20"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M5.84 15.63a6.97 6.97 0 008.32 0 8.2 8.2 0 00-8.32 0zM4.7 14.57a7 7 0 1110.6 0 9.7 9.7 0 00-10.6 0zM10 1.5a8.5 8.5 0 100 17 8.5 8.5 0 000-17zm-1.5 7a1.5 1.5 0 103 0 1.5 1.5 0 00-3 0zm1.5-3a3 3 0 100 6 3 3 0 000-6z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
            <span>
              My profile
            </span>
          </li>
          <li
            class="item"
          >
            <svg
              fill="none"
              height="20"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M11.84 2H8.16c-.93 0-1.67 0-2.26.05-.62.05-1.15.15-1.63.4a4.15 4.15 0 00-1.82 1.82 4.26 4.26 0 00-.4 1.63C2 6.5 2 7.23 2 8.16v3.68c0 .93 0 1.67.05 2.26.05.62.15 1.15.4 1.63.4.78 1.04 1.42 1.82 1.82.48.25 1.01.35 1.63.4.6.05 1.33.05 2.26.05h3.68c.93 0 1.67 0 2.26-.05a4.26 4.26 0 001.63-.4 4.15 4.15 0 001.82-1.82c.25-.48.35-1.01.4-1.63.05-.6.05-1.33.05-2.26V8.16c0-.93 0-1.67-.05-2.26a4.26 4.26 0 00-.4-1.63 4.15 4.15 0 00-1.82-1.82 4.26 4.26 0 00-1.63-.4C13.5 2 12.77 2 11.84 2zm-6.9 1.79c.25-.12.56-.2 1.08-.25.53-.04 1.2-.04 2.17-.04h3.62c.96 0 1.64 0 2.17.04.52.05.83.13 1.07.25.5.25.9.66 1.16 1.16.12.24.2.55.25 1.07l.02.48H3.52l.02-.48c.05-.52.13-.83.25-1.07.25-.5.66-.9 1.16-1.16zM3.5 8v3.81c0 .96 0 1.64.04 2.17.05.52.13.83.25 1.07.25.5.66.9 1.16 1.16.24.12.55.2 1.07.25.53.04 1.2.04 2.17.04h3.62c.96 0 1.64 0 2.17-.04a2.8 2.8 0 001.07-.25c.5-.25.9-.66 1.16-1.16.12-.24.2-.55.25-1.07.04-.53.04-1.2.04-2.17V8.19 8z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
            <span>
              News
            </span>
          </li>
          <li
            class="item"
          >
            <svg
              height="20"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
                id="message_outline_20__Page-2"
                stroke="none"
                stroke-width="1"
              >
                <g
                  id="message_outline_20__message_outline_20"
                >
                  <path
                    d="M0 0h20v20H0z"
                    id="message_outline_20__Shape"
                    opacity=".4"
                  />
                  <path
                    d="M6.83 15.75c.2-.23.53-.31.82-.2.81.3 1.7.45 2.6.45 3.77 0 6.75-2.7 6.75-6s-2.98-6-6.75-6S3.5 6.7 3.5 10c0 1.21.4 2.37 1.14 3.35.1.14.16.31.15.49-.04.76-.4 1.78-1.08 3.13 1.48-.11 2.5-.53 3.12-1.22zM3.24 18.5a1.2 1.2 0 01-1.1-1.77A10.77 10.77 0 003.26 14 7 7 0 012 10c0-4.17 3.68-7.5 8.25-7.5S18.5 5.83 18.5 10s-3.68 7.5-8.25 7.5c-.92 0-1.81-.13-2.66-.4-1 .89-2.46 1.34-4.35 1.4z"
                    fill="currentColor"
                    fill-rule="nonzero"
                    id="message_outline_20__Icon-Color"
                  />
                </g>
              </g>
            </svg>
            <span>
              Messages
            </span>
          </li>
          <li
            class="item"
          >
            <svg
              height="20"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
                id="phone_outline_20__Icons"
                stroke="none"
                stroke-width="1"
              >
                <g
                  id="phone_outline_20__Icons-20/phone_outline_20"
                >
                  <g
                    id="phone_outline_20__phone_outline_20"
                  >
                    <path
                      d="M0 0h20v20H0z"
                      opacity=".1"
                    />
                    <path
                      d="M7.78 2.7l.7.83a4.13 4.13 0 01-.3 5.58l-.3.3c.23.38.64.88 1.23 1.48.53.53.99.9 1.35 1.14l.13.09.3-.3a4.14 4.14 0 015.58-.3l.83.7a3.45 3.45 0 01.34 4.9c-.6.69-1.43 1.15-2.29 1.28l-.5.06c-3 .3-6.04-1.16-9.1-4.21-3.05-3.06-4.5-6.1-4.22-9.03l.06-.51a3.83 3.83 0 011.3-2.36l.17-.13a3.44 3.44 0 014.72.48zm-3.91.79a2.38 2.38 0 00-.81 1.58c-.35 2.55.9 5.25 3.76 8.12 2.86 2.86 5.56 4.1 8.12 3.75.6-.05 1.17-.35 1.57-.8.72-.85.64-2.09-.18-2.77l-.83-.7-.14-.12a2.64 2.64 0 00-3.41.33l-.6.6-.09.07a.9.9 0 01-.86.14 7.26 7.26 0 01-2.35-1.74A7.23 7.23 0 016.31 9.6a.9.9 0 01.21-.95l.6-.6.13-.13c.86-1 .9-2.44.08-3.42l-.7-.83-.11-.13c-.7-.7-1.85-.73-2.65-.05z"
                      fill="currentColor"
                      fill-rule="nonzero"
                      id="phone_outline_20__Icon-Color"
                    />
                  </g>
                </g>
              </g>
            </svg>
            <span>
              Calls
            </span>
          </li>
          <li
            class="item"
          >
            <svg
              fill="none"
              height="20"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="currentColor"
              >
                <g
                  clip-rule="evenodd"
                  fill-rule="evenodd"
                >
                  <path
                    d="M6.25 3.5a3 3 0 100 6 3 3 0 000-6zm-1.5 3a1.5 1.5 0 103 0 1.5 1.5 0 00-3 0zM2.69 11.57c.96-.55 2.22-.82 3.56-.82s2.6.27 3.56.82c.98.56 1.69 1.47 1.69 2.68 0 .7-.28 1.3-.78 1.71-.48.39-1.1.54-1.72.54H3.5c-.61 0-1.24-.15-1.72-.54-.5-.4-.78-1-.78-1.71 0-1.21.71-2.12 1.69-2.68zm.75 1.3c-.65.37-.94.84-.94 1.38 0 .3.1.44.22.54.14.11.4.21.78.21H9c.39 0 .64-.1.78-.21.12-.1.22-.25.22-.54 0-.54-.29-1-.94-1.38-.66-.39-1.65-.62-2.81-.62s-2.15.23-2.81.62zM13.75 3.5a3 3 0 100 6 3 3 0 000-6zm-1.5 3a1.5 1.5 0 103 0 1.5 1.5 0 00-3 0z"
                  />
                </g>
                <path
                  d="M13.75 12.25c-.23 0-.45.01-.68.03a.75.75 0 11-.14-1.49c.27-.03.54-.04.82-.04 1.34 0 2.6.27 3.56.82.98.56 1.69 1.47 1.69 2.68 0 .7-.28 1.3-.78 1.71-.48.39-1.1.54-1.72.54h-3a.75.75 0 010-1.5h3c.39 0 .64-.1.78-.21.12-.1.22-.25.22-.54 0-.54-.29-1-.94-1.38a5.77 5.77 0 00-2.81-.62z"
                />
              </g>
            </svg>
            <span>
              Friends
            </span>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar"
    >
      <ul>
        <li
          class="item"
        >
          <svg
            fill="none"
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.84 15.63a6.97 6.97 0 008.32 0 8.2 8.2 0 00-8.32 0zM4.7 14.57a7 7 0 1110.6 0 9.7 9.7 0 00-10.6 0zM10 1.5a8.5 8.5 0 100 17 8.5 8.5 0 000-17zm-1.5 7a1.5 1.5 0 103 0 1.5 1.5 0 00-3 0zm1.5-3a3 3 0 100 6 3 3 0 000-6z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span>
            My profile
          </span>
        </li>
        <li
          class="item"
        >
          <svg
            fill="none"
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.84 2H8.16c-.93 0-1.67 0-2.26.05-.62.05-1.15.15-1.63.4a4.15 4.15 0 00-1.82 1.82 4.26 4.26 0 00-.4 1.63C2 6.5 2 7.23 2 8.16v3.68c0 .93 0 1.67.05 2.26.05.62.15 1.15.4 1.63.4.78 1.04 1.42 1.82 1.82.48.25 1.01.35 1.63.4.6.05 1.33.05 2.26.05h3.68c.93 0 1.67 0 2.26-.05a4.26 4.26 0 001.63-.4 4.15 4.15 0 001.82-1.82c.25-.48.35-1.01.4-1.63.05-.6.05-1.33.05-2.26V8.16c0-.93 0-1.67-.05-2.26a4.26 4.26 0 00-.4-1.63 4.15 4.15 0 00-1.82-1.82 4.26 4.26 0 00-1.63-.4C13.5 2 12.77 2 11.84 2zm-6.9 1.79c.25-.12.56-.2 1.08-.25.53-.04 1.2-.04 2.17-.04h3.62c.96 0 1.64 0 2.17.04.52.05.83.13 1.07.25.5.25.9.66 1.16 1.16.12.24.2.55.25 1.07l.02.48H3.52l.02-.48c.05-.52.13-.83.25-1.07.25-.5.66-.9 1.16-1.16zM3.5 8v3.81c0 .96 0 1.64.04 2.17.05.52.13.83.25 1.07.25.5.66.9 1.16 1.16.24.12.55.2 1.07.25.53.04 1.2.04 2.17.04h3.62c.96 0 1.64 0 2.17-.04a2.8 2.8 0 001.07-.25c.5-.25.9-.66 1.16-1.16.12-.24.2-.55.25-1.07.04-.53.04-1.2.04-2.17V8.19 8z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span>
            News
          </span>
        </li>
        <li
          class="item"
        >
          <svg
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
              id="message_outline_20__Page-2"
              stroke="none"
              stroke-width="1"
            >
              <g
                id="message_outline_20__message_outline_20"
              >
                <path
                  d="M0 0h20v20H0z"
                  id="message_outline_20__Shape"
                  opacity=".4"
                />
                <path
                  d="M6.83 15.75c.2-.23.53-.31.82-.2.81.3 1.7.45 2.6.45 3.77 0 6.75-2.7 6.75-6s-2.98-6-6.75-6S3.5 6.7 3.5 10c0 1.21.4 2.37 1.14 3.35.1.14.16.31.15.49-.04.76-.4 1.78-1.08 3.13 1.48-.11 2.5-.53 3.12-1.22zM3.24 18.5a1.2 1.2 0 01-1.1-1.77A10.77 10.77 0 003.26 14 7 7 0 012 10c0-4.17 3.68-7.5 8.25-7.5S18.5 5.83 18.5 10s-3.68 7.5-8.25 7.5c-.92 0-1.81-.13-2.66-.4-1 .89-2.46 1.34-4.35 1.4z"
                  fill="currentColor"
                  fill-rule="nonzero"
                  id="message_outline_20__Icon-Color"
                />
              </g>
            </g>
          </svg>
          <span>
            Messages
          </span>
        </li>
        <li
          class="item"
        >
          <svg
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
              id="phone_outline_20__Icons"
              stroke="none"
              stroke-width="1"
            >
              <g
                id="phone_outline_20__Icons-20/phone_outline_20"
              >
                <g
                  id="phone_outline_20__phone_outline_20"
                >
                  <path
                    d="M0 0h20v20H0z"
                    opacity=".1"
                  />
                  <path
                    d="M7.78 2.7l.7.83a4.13 4.13 0 01-.3 5.58l-.3.3c.23.38.64.88 1.23 1.48.53.53.99.9 1.35 1.14l.13.09.3-.3a4.14 4.14 0 015.58-.3l.83.7a3.45 3.45 0 01.34 4.9c-.6.69-1.43 1.15-2.29 1.28l-.5.06c-3 .3-6.04-1.16-9.1-4.21-3.05-3.06-4.5-6.1-4.22-9.03l.06-.51a3.83 3.83 0 011.3-2.36l.17-.13a3.44 3.44 0 014.72.48zm-3.91.79a2.38 2.38 0 00-.81 1.58c-.35 2.55.9 5.25 3.76 8.12 2.86 2.86 5.56 4.1 8.12 3.75.6-.05 1.17-.35 1.57-.8.72-.85.64-2.09-.18-2.77l-.83-.7-.14-.12a2.64 2.64 0 00-3.41.33l-.6.6-.09.07a.9.9 0 01-.86.14 7.26 7.26 0 01-2.35-1.74A7.23 7.23 0 016.31 9.6a.9.9 0 01.21-.95l.6-.6.13-.13c.86-1 .9-2.44.08-3.42l-.7-.83-.11-.13c-.7-.7-1.85-.73-2.65-.05z"
                    fill="currentColor"
                    fill-rule="nonzero"
                    id="phone_outline_20__Icon-Color"
                  />
                </g>
              </g>
            </g>
          </svg>
          <span>
            Calls
          </span>
        </li>
        <li
          class="item"
        >
          <svg
            fill="none"
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="currentColor"
            >
              <g
                clip-rule="evenodd"
                fill-rule="evenodd"
              >
                <path
                  d="M6.25 3.5a3 3 0 100 6 3 3 0 000-6zm-1.5 3a1.5 1.5 0 103 0 1.5 1.5 0 00-3 0zM2.69 11.57c.96-.55 2.22-.82 3.56-.82s2.6.27 3.56.82c.98.56 1.69 1.47 1.69 2.68 0 .7-.28 1.3-.78 1.71-.48.39-1.1.54-1.72.54H3.5c-.61 0-1.24-.15-1.72-.54-.5-.4-.78-1-.78-1.71 0-1.21.71-2.12 1.69-2.68zm.75 1.3c-.65.37-.94.84-.94 1.38 0 .3.1.44.22.54.14.11.4.21.78.21H9c.39 0 .64-.1.78-.21.12-.1.22-.25.22-.54 0-.54-.29-1-.94-1.38-.66-.39-1.65-.62-2.81-.62s-2.15.23-2.81.62zM13.75 3.5a3 3 0 100 6 3 3 0 000-6zm-1.5 3a1.5 1.5 0 103 0 1.5 1.5 0 00-3 0z"
                />
              </g>
              <path
                d="M13.75 12.25c-.23 0-.45.01-.68.03a.75.75 0 11-.14-1.49c.27-.03.54-.04.82-.04 1.34 0 2.6.27 3.56.82.98.56 1.69 1.47 1.69 2.68 0 .7-.28 1.3-.78 1.71-.48.39-1.1.54-1.72.54h-3a.75.75 0 010-1.5h3c.39 0 .64-.1.78-.21.12-.1.22-.25.22-.54 0-.54-.29-1-.94-1.38a5.77 5.77 0 00-2.81-.62z"
              />
            </g>
          </svg>
          <span>
            Friends
          </span>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
